clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -764,24 +770,36 @@ static std::string computeUniqueSYCLXOCCName(StringRef Name,
     Result.push_back(c);
   }
 
+  // Replace first kernel character name by a 'k' to be compatible with SPIR
+  if (!Result.empty() && (Result.front() == '_' || isDigit(Result.front()))) {
```

Comment:
By looking at all these `Result.empty()` everywhere I wonder in which case this could be empty.